### DISCUSSION TITLE
"builder error for url" when creating an instance of a DeltaTable which is located in an azurite blob storage

### DISCUSSION THREAD

**Discussion by ghost** [delta-io/delta-rs#2901] [Q&A] [answered]
# Environment

**Delta-rs version**: 0.19.1

**Environment**:
- **Cloud provider**: Azurite
- **OS**: Mac OS

***
# Bug

**What happened**:
When trying to create an instance of a DeltaTable object, I get following error: 
"Generic MicrosoftAzure error: Error after 0 retries in 9.625µs, max_retries:10, retry_timeout:180s, source:builder error for url (http://localhost:10000/landing-zone-files/produktfrageboegenLandingZone/0.0.1/eQMAllgemeineArtikelInfos/_delta_log/_last_checkpoint)"

**What you expected to happen**:
I would expect that the instance of the DataTable has been created and I am able to query or write into the table.

**How to reproduce it**:
I use following code to create an instance of a DeltaTable: 
```
storage_options = {
                "endpoint": "http://localhost:10000",
                 "account_name": "devstoreaccount1",
                "account_key": "ACCOUNT_KEY",
                "default_endpoints_protocol": "http"
            }
delta_table = DeltaTable(f"az://LandingZone/DataContractName/0.0.1/ModelName", storage_options=storage_options)
```

**Comment by g0di** [comment, marked as answer]
Hello, I found out that you actually have to pass a dedicated option for using an Azurite emulated storage account container.

You can do it either from environment variables
```bash
AZURE_STORAGE_USE_EMULATOR="1"
```

Or directly in source code
```python
storage_options = { "azure_storage_use_emulator": "1" }
delta_table = DeltaTable(f"az://LandingZone/DataContractName/0.0.1/ModelName", storage_options=storage_options)
```
